src/core/connect-extension/conn-ext-build-state/build-tokens.ts

```
@@ -39,21 +36,27 @@ const fetchToken = async (
     let blockchainToken;
 
     try {
-        const fetchResponse = await fetch(
-            CONFIG.tokensUrl +
-                `${blockchain.toLocaleLowerCase()}/${tk.symbol.toLocaleLowerCase()}.json`
-        );
-
-        if (fetchResponse.status === 200) {
-            staticToken = await fetchResponse.json();
-            blockchainToken = await getBlockchain(blockchain)
+        const getTokenInfo = await Promise.all([
+            fetch(
+                CONFIG.tokensUrl +
+                    `${blockchain.toLocaleLowerCase()}/${tk.symbol.toLocaleLowerCase()}.json`
+            ),
+            getBlockchain(blockchain)
                 .getClient(chainId)
-                .tokens[tk.type].getTokenInfo(tk.contractAddress);
+                .tokens[tk.type].getTokenInfo(tk.contractAddress)
+        ]);
+        if (getTokenInfo[0].status === 200) {
+            staticToken = await getTokenInfo[0].json();
         }
+        blockchainToken = getTokenInfo[1];
     } catch {
         //
```

From now on no more empty catches :) Let's take an action if something fails.